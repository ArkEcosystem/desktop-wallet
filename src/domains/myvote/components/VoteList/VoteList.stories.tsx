import React from "react";

import { VoteList } from "./VoteList";

export default { title: "My Votes / Components / VoteList" };

const votes = [
	{
		address: "FJKDSALJFKASLJFKSDAJD333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 1",
	},
	{
		address: "AhFJKDSALJFKASLJFKSDEAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 2",
	},
	{
		address: "FAhFJKDSALJFKASLJFKSFDAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 3",
	},
];

export const Default = () => <VoteList votes={votes} />;
