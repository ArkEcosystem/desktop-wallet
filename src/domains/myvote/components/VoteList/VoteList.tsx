import { Address } from "app/components/Address";
import { TransactionDetail } from "app/components/TransactionDetail";
import { Avatar } from "domains/wallet/components/Avatar";
import React from "react";

type VoteListProps = {
	votes?: any[];
};

export const VoteList = ({ votes }: VoteListProps) => {
	return (
		<div className="-my-5">
			{votes?.map((vote: any, index: number) => (
				<TransactionDetail key={index} border={index !== 0} label=" " extra={<Avatar address={vote.address} />}>
					<Address address={vote.address} walletName={vote.delegateName} size="large" />
				</TransactionDetail>
			))}
		</div>
	);
};

VoteList.defaultProps = {
	votes: [],
};
