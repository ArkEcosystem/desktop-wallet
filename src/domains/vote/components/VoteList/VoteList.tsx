import { Address } from "app/components/Address";
import { Avatar } from "app/components/Avatar";
import { TransactionDetail } from "app/components/TransactionDetail";
import React from "react";

type VoteListProps = {
	votes?: any[];
};

export const VoteList = ({ votes }: VoteListProps) => {
	return (
		<div className="-my-5">
			{votes?.map((vote: any, index: number) => (
				<TransactionDetail key={index} border={index !== 0} extra={<Avatar size="lg" address={vote.address} />}>
					<Address address={vote.address} walletName={vote.delegateName} />
				</TransactionDetail>
			))}
		</div>
	);
};

VoteList.defaultProps = {
	votes: [],
};
