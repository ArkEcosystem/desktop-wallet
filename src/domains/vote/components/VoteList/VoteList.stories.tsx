import React from "react";

import { VoteList } from "./VoteList";

export default { title: "Domains / Vote / Components / VoteList" };

const votes = [
	{
		address: "FJKDSALJFKASLJFKSDAJD333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 1",
	},
	{
		address: "AhFJKDSALJFKASLJFKSDEAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 2",
	},
	{
		address: "FAhFJKDSALJFKASLJFKSFDAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 3",
	},
];

export const Default = () => <VoteList votes={votes} />;
