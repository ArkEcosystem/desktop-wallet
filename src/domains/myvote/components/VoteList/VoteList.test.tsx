import { render } from "@testing-library/react";
import React from "react";

import { VoteList } from "./VoteList";

const votes = [
	{
		address: "FJKDSALJFKASLJFKSDAJD333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 1",
	},
	{
		address: "AhFJKDSALJFKASLJFKSDEAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 2",
	},
	{
		address: "FAhFJKDSALJFKASLJFKSFDAJ333FKFKDSAJFKSAJFKLASJKDFJ",
		delegateName: "Delegate 3",
	},
];

describe("VoteList", () => {
	it("should render", () => {
		const { container, asFragment } = render(<VoteList votes={votes} />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with empty list", () => {
		const { container, asFragment } = render(<VoteList votes={[]} />);

		expect(container).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});
});
