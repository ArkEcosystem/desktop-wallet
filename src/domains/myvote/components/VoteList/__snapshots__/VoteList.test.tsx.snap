// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VoteList should render 1`] = `
<DocumentFragment>
  <div
    class="-my-5"
  >
    <div
      class="flex items-center py-4 true "
    >
      <div
        class="flex-1"
      >
        <div
          class="text-sm font-semibold text-theme-neutral-500"
        >
           
        </div>
        <div
          class="mt-2 font-semibold"
        >
          <div
            class="inline-block truncate"
          >
            <span
              class="text-theme-neutral-800 font-semibold max-w-24 flex-auto truncate mt-4 mr-1 text-xl"
              data-testid="address__wallet-name"
            >
              Delegate 1
            </span>
            <span
              class="text-theme-neutral-400 font-semibold text-xl"
              data-testid="address__wallet-address"
            >
              FJKDSA...SJKDFJ
            </span>
          </div>
        </div>
      </div>
      <div
        class="Avatar__AvatarStyled-yiil9n-0 fympLW"
        data-testid="Avatar"
      />
    </div>
    <div
      class="flex items-center py-4 border-t border-dashed border-theme-neutral-300 "
    >
      <div
        class="flex-1"
      >
        <div
          class="text-sm font-semibold text-theme-neutral-500"
        >
           
        </div>
        <div
          class="mt-2 font-semibold"
        >
          <div
            class="inline-block truncate"
          >
            <span
              class="text-theme-neutral-800 font-semibold max-w-24 flex-auto truncate mt-4 mr-1 text-xl"
              data-testid="address__wallet-name"
            >
              Delegate 2
            </span>
            <span
              class="text-theme-neutral-400 font-semibold text-xl"
              data-testid="address__wallet-address"
            >
              AhFJKD...SJKDFJ
            </span>
          </div>
        </div>
      </div>
      <div
        class="Avatar__AvatarStyled-yiil9n-0 fympLW"
        data-testid="Avatar"
      />
    </div>
    <div
      class="flex items-center py-4 border-t border-dashed border-theme-neutral-300 "
    >
      <div
        class="flex-1"
      >
        <div
          class="text-sm font-semibold text-theme-neutral-500"
        >
           
        </div>
        <div
          class="mt-2 font-semibold"
        >
          <div
            class="inline-block truncate"
          >
            <span
              class="text-theme-neutral-800 font-semibold max-w-24 flex-auto truncate mt-4 mr-1 text-xl"
              data-testid="address__wallet-name"
            >
              Delegate 3
            </span>
            <span
              class="text-theme-neutral-400 font-semibold text-xl"
              data-testid="address__wallet-address"
            >
              FAhFJK...SJKDFJ
            </span>
          </div>
        </div>
      </div>
      <div
        class="Avatar__AvatarStyled-yiil9n-0 fympLW"
        data-testid="Avatar"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`VoteList should render with empty list 1`] = `
<DocumentFragment>
  <div
    class="-my-5"
  />
</DocumentFragment>
`;
